tests/test_source_post_voice_history_live.py

```
@@ -0,0 +1,576 @@
+)
+
+
+def _database_dsn(database_name: str) -> str:
+    parsed = urlsplit(_ADMIN_DSN)
+    return urlunsplit(parsed._replace(path=f"/{database_name}"))
+
+
+def _apply_migrations(database_dsn: str) -> None:
+    """Replay every numbered migration through psql, matching migrate.sh."""
+    for migration in sorted(_MIGRATIONS_DIR.glob("*.sql")):
+        subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)